src/pds_doi_service/core/actions/test/check_test.py

```diff
@@ -221,42 +204,20 @@ def test_email_receipt(self):
         # Create a new check action so our patched config is pulled in
         action = DOICoreActionCheck(self.db_name)
 
-        with tempfile.TemporaryFile() as temp_file:
-            # Stand up a subprocess running a debug smtpd server
-            # By default, all this server is does is echo email payloads to
-            # standard out, so provide a temp file to capture it
-            debug_email_proc = subprocess.Popen(
-                ["python", "-u", "-m", "smtpd", "-n", "-c", "DebuggingServer", "localhost:1025"], stdout=temp_file
-            )
-
-            # Give the debug smtp server a chance to start listening
-            time.sleep(1)
-
-            try:
-                # Run the check action and have it send an email w/ attachment
-                action.run(email=True, attachment=True, submitter="email-test@email.com")
-
-                # Read the raw email contents (payload) from the subprocess
-                # into a string
-                temp_file.seek(0)
-                email_contents = temp_file.read()
-                message = message_from_bytes(email_contents).get_payload()
-            finally:
-                # Send the debug smtp server a ctrl+C and wait for it to stop
-                os.kill(debug_email_proc.pid, signal.SIGINT)
-                debug_email_proc.wait()
-
-        # Run some string searches on the email body to ensure what we expect
-        # made it in
-
-        # Email address provided to check action should be present
-        self.assertIn("email-test@email.com", message)
+        message = capture_email(lambda: action.run(email=True, attachment=True, submitter="email-test@email.com"))
+        html_part, attachment_part = message.get_payload()
+        attachment = attachment_part.get_payload()[0].get_payload()[0]
+
+        # Email address provided to check action should be present in recipients
+        recipients = message["To"].strip().split(", ")
+        self.assertIn("email-test@email.com", recipients)
```

Comment:
Oh! Looks like the `Popen` of `python` is gone anyway! Well for the future, `sys.executable` is there and ready for yo.

This way of capturing the email without the messy and/or brittle subprocess handling is definitely better 👍